### DISCUSSION TITLE
pyCMPL on Raspberry Pi

### DISCUSSION THREAD

**Discussion by 356255531** [MikeSteglich/Cmpl2#12] [Q&A]
Dear Authors,

I am aware of the dedicated CMPL 1.12 version for Raspberry Pi. Is there also a similar release for CMPL 2.X with python3? Although the 2.0.1 is claimed to be compatible for Raspbian in (here)[http://www.[coliop.org](http://www.coliop.org/index.html)/index.html]

I tried with the official release of  CMPL 2.0 .1binary and pyCmpl on my Raspberry Pi 2 and ended up with a exec format error, which indicated the CPU architecture incompatibility. 

Therefore, I also tried to compile the CMPL 2.0 from source on PI, but the pyCmpl solver still encountered with a Cmpl Exception. The error log is as following:

>  File "/best_components/labinternalcomponents/utils/utils.py", line 250, in action_handler
    update_handler(simulator_info, pre_jsons)
  File "/best_components/labinternalcomponents/obsd/obsd.py", line 463, in update
    scheduler.schedule(
  File "/best_components/labinternalcomponents/obsd/scheduler.py", line 128, in schedule
    m.solve()
  File "/best_components/labinternalcomponents/resources/cmpl/pyCmpl/lib3/pyCmpl/Cmpl.py", line 800, in solve
    self.__runCmpl(cmdList)
  File "/best_components/labinternalcomponents/resources/cmpl/pyCmpl/lib3/pyCmpl/Cmpl.py", line 1152, in __runCmpl
    raise CmplException("Cmpl finished with errors", self.__status.cmplMessageList)
pyCmpl.CmplException.CmplException: ERROR:

It seems that the CMPL binary can be called now, but somehow still problematic. I don't know if the problem is still with the CPU architecture or something else. Do you have any comment on how can I progress or at least debug?

Many thanks,
Zhiwei

**Comment by MikeSteglich** [comment]
Hello Zhiwei,

Compiling and packaging on Raspi is a bit tricky. Are you using Ubuntu on Arm or Rasbian? I could send an unofficial version made on Ubuntu tomorrow or I could compile it on Rasbian.

Thanks,

Mike

**Reply by 356255531** [reply]
> Hello Zhiwei,
> 
> Compiling and packaging on Raspi is a bit tricky. Are you using Ubuntu on Arm or Rasbian? I could send an unofficial version made on Ubuntu tomorrow or I could compile it on Rasbian.
> 
> Thanks,
> 
> Mike

Dear Prof. Steglich,

I think I was wrong with the OS version. Currently I am using a customized  PI from the third-party and the system info is:
Static hostname: BESTlabExternalPi
Icon name: computer
Machine ID: 48c5f4b3982345058855faa0a2da59a3
Boot ID: ac6f9441d120432382ced8427403f65b
Operating System: Debian GNU/Linux 11 (bullseye)
Kernel: Linux 6.1.19-v8+
Architecture: arm64

Apparently we are not running Raspbian but Debian instead, however, Raspbian is based on Debian. In short, an arm CPU + Debian. I don't know if it would be possible for you to compile the CMPL2 with that combo.  But we did it by running 
```
make Cmpl
```
in the project root but end up with the above error.

**Reply by MikeSteglich** [reply]
Could you please use the binary I sent yesterday. It was created on Ubuntu on Arm and should run on your system.
